Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -694,7 +704,15 @@ private class ModuleAliasTracker {
                   target: String,
                   product: String,
                   originPackage: PackageIdentity,
-                  consumingPackage: PackageIdentity) {
+                  consumingPackage: PackageIdentity) throws {
+        if let aliasDict = aliasMap[originPackage] {
+            let models = aliasDict.values.flatMap{$0}.filter { $0.name == target }
+            if !models.isEmpty {
+                // Error if there are multiple aliases specified for this product dependency
+                throw PackageGraphError.multipleModuleAliases(target: target, product: product, package: originPackage.description, aliases: models.map{$0.alias} + [alias])
```

Comment:
If I had more than one conflicting alias, I'd see an error about one at a time, right?  I think that's fine, just checking if I understand the flow.

Reply:
this error will show all conflicting aliases defined for that target in the product